File: Kyoo.Common/Models/Episode.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace Kyoo.Models
{
	public class Episode
	{
		[JsonIgnore] public long ID { get; set; }
		[JsonIgnore] public long ShowID { get; set; }
		[JsonIgnore] public virtual Show Show { get; set; }
		[JsonIgnore] public long? SeasonID { get; set; }
		[JsonIgnore] public virtual Season Season { get; set; }

		public long SeasonNumber { get; set; }
		public long EpisodeNumber { get; set; }
		public long AbsoluteNumber { get; set; }
		[JsonIgnore] public string Path { get; set; }
		public string Title { get; set; }
		public string Overview { get; set; }
		public DateTime? ReleaseDate { get; set; }

		public long Runtime { get; set; } //This runtime variable should be in minutes

		[JsonIgnore] public string ImgPrimary { get; set; }
		public virtual IEnumerable<MetadataID> ExternalIDs { get; set; }

		[JsonIgnore] public virtual IEnumerable<Track> Tracks { get; set; }

		public string ShowTitle => Show.Title; // Used in the API response only
		public string Slug => GetSlug(Show.Slug, SeasonNumber, EpisodeNumber);
		public string Thumb
		{
			get
			{
				if (Show != null)
					return "thumb/" + Slug;
				return ImgPrimary;
			}
		}


		public Episode() { }

		public Episode(long seasonNumber, 
			long episodeNumber,
			long absoluteNumber,
			string title,
			string overview,
			DateTime? releaseDate,
			long runtime,
			string imgPrimary,
			IEnumerable<MetadataID> externalIDs)
		{
			SeasonNumber = seasonNumber;
			EpisodeNumber = episodeNumber;
			AbsoluteNumber = absoluteNumber;
			Title = title;
			Overview = overview;
			ReleaseDate = releaseDate;
			Runtime = runtime;
			ImgPrimary = imgPrimary;
			ExternalIDs = externalIDs;
		}

		public Episode(long showID, 
			long seasonID,
			long seasonNumber, 
			long episodeNumber, 
			long absoluteNumber, 
			string path,
			string title, 
			string overview, 
			DateTime? releaseDate, 
			long runtime, 
			string imgPrimary,
			IEnumerable<MetadataID> externalIDs)
		{
			ShowID = showID;
			SeasonID = seasonID;
			SeasonNumber = seasonNumber;
			EpisodeNumber = episodeNumber;
			AbsoluteNumber = absoluteNumber;
			Path = path;
			Title = title;
			Overview = overview;
			ReleaseDate = releaseDate;
			Runtime = runtime;
			ImgPrimary = imgPrimary;
			ExternalIDs = externalIDs;
		}

		public static string GetSlug(string showSlug, long seasonNumber, long episodeNumber)
		{
			return showSlug + "-s" + seasonNumber + "e" + episodeNumber;
		}
	}
}

File: scanner/providers/types/episode.py
import os
from datetime import date
from dataclasses import dataclass, field, asdict
from typing import Optional

from .show import Show
from .metadataid import MetadataID


@dataclass
class PartialShow:
	name: str
	original_language: Optional[str]
	external_id: dict[str, MetadataID]


@dataclass
class EpisodeID:
	show_id: str
	season: Optional[int]
	episode: int
	link: Optional[str]


@dataclass
class EpisodeTranslation:
	name: str
	overview: Optional[str]


@dataclass
class Episode:
	show: Show | PartialShow
	season_number: int
	episode_number: int
	absolute_number: int
	runtime: Optional[int]
	release_date: Optional[date | int]
	thumbnail: Optional[str]
	external_id: dict[str, EpisodeID]

	path: Optional[str] = None
	show_id: Optional[str] = None
	season_id: Optional[str] = None
	translations: dict[str, EpisodeTranslation] = field(default_factory=dict)

	def to_kyoo(self):
		# For now, the API of kyoo only support one language so we remove the others.
		default_language = os.environ["LIBRARY_LANGUAGES"].split(",")[0]
		return {
			**asdict(self),
			**asdict(self.translations[default_language]),
			"show": None,
		}
